src/server/ua_services_attribute.c

```diff
@@ -118,7 +119,16 @@ readValueAttributeFromNode(UA_Server *server, UA_Session *session,
                                        session->sessionHandle, &vn->nodeId,
                                        vn->context, rangeptr, &vn->value.data.value);
         UA_LOCK(server->serviceMutex);
+#ifdef UA_ENABLE_USE_ENCODED_NODES
+        if(server->isServerListening == UA_TRUE) {
```

Comment:
No.

This is not the place to decide whether a node is in the binary format.
This has to be done **inside the nodestore plugin**. It should be totally opaque to the server.